### ISSUE TITLE
settings window proposal

### ISSUE THREAD

**Issue by jessicajonesunitedkingdom** [emoon/HippoPlayer#113]
Hi

Look at screenshot of foobar2000 very handy settings window.
Maybe you want to make it like this? According to me this is the best way to do it.

A treeview at the left with categories and at the right the selected category with all its options

I also show this because I would love to have a "plugins" section (see this screenshot) which lists the plugins and shows the version of the LIBRARY and not the plugin's version (as seen here in foobar2000).
So it would be great to see for example: adplug plugin, version: 2.3.3 etc
This way we can see if all plugins are up to date easily, by displaying the version number of the library used.

What I also like is the winamp way to configure a plugin: double click it for settings of that plugin.
(here they add each plugin's setting in the treeview)

![settings](https://user-images.githubusercontent.com/65877513/87802825-08478600-c852-11ea-8555-e7cf84d087ff.jpg)

**Comment by emoon** [comment]
Yeah I have seen this settings layout in foobar2000 and I will likely make something similar. When it comes to version I think it's good to have a "Version" and "Library Version". The reason I think that is good is because there may be a fix in the plugin so the version of the plugin goes up but the library remains the same.

**Comment by jessicajonesunitedkingdom** [comment]
"Version" and "Library Version is even better!

I would also like very much a checkbox in front of each plugin: in this way we can force the player to use a specific plugin to play files that can be played be more then 1 plugin.

This can be seen in audacious, see screenshot.

![audacious](https://user-images.githubusercontent.com/65877513/87804182-cc152500-c853-11ea-8c8c-d46d6d1c9762.jpg)

**Comment by emoon** [comment]
But is the screenshot above a global setting or a "per file setting" ? Something I want to have is that you can right click on a file and say "use this player x" (and this can be overriding a directory of files also for example)

**Comment by jessicajonesunitedkingdom** [comment]
This is a global setting. If you disable a plugin in audacious, ALL files that need that plugin can NOT be played (until you enable it again).
In foobar2000 you must uninstall a plugin to disable it (not good!).

Your idea seems a bit complicated. I don't think users will need such an advanced setting. I would personally not need it (global is fine for me). But maybe other people would want to use it. Where will you all store settings for EACH file?
It could introduce much bugs IMHO (compared to global setting)

**Comment by emoon** [comment]
The idea is if there are several players that can play a certain file but if one of them plays it wrong / with less quality you can override. This is something that XMPlay support for example and has been requested https://github.com/emoon/HippoPlayer/issues/50

**Comment by emoon** [comment]
It's also so you can tweak settings for an specific song and the player used (as seen in #50)

**Comment by jessicajonesunitedkingdom** [comment]
I didn't think that way, it's very unique feature!
Not much players have it

**Comment by emoon** [comment]
Indeed. I think very few players allow you to configure it this way. Also these settings can be saved in the database. I'm not exactly sure how to approach it just yet, but then it's a bit down on the priority list as I need to get the general settings in first.

**Comment by emoon** [comment]
Making some progress on this. This is currently how it looks. It's not done yet, but getting there.

![hippo_plugin_settings](https://user-images.githubusercontent.com/350822/103435970-f9de0280-4c16-11eb-8fa2-e46e82fc12a5.png)
